### PR TITLE
PIMS-677: Create endpoint for LTSA API

### PR DESCRIPTION
**Pull request by LawrenceLau2020**
![DESCRIPTION]

This is still a work in progress for the new LTSA api for the backend which is close to 100% complete. The calls to authenticate with LTSA and to obtain the LTSA summary and the LTSA order to get the title information have been created and tested locally.
You can test this using Thunderclient using this endpoint: ~~http://localhost:8081/api/properties/parcels/ltsa/auth?pid=xxxxxxxxx~~ http://localhost:8081/api/ltsa/land/title?pid=xxxxxxxxx

There are some new backend unit tests for the new methods related to the LTSA api. see [PIMS-685](https://apps.itsm.gov.bc.ca/jira/browse/PIMS-685) and all the new backend unit tests are currently passing.

In order for it to work locally, you will need to add the following to your /backend/api/.env file (just replace the **** with the actual credentials), and also need to change the **ASPNETCORE_ENVIRONMENT** variable to **Local**.
```
# LTSA Credentials
Ltsa__IntegratorUsername=*****
Ltsa__IntegratorPassword=*****
Ltsa__UserName=*****
Ltsa__UserPassword=*****
```

<br />

![TICKETS_AND_ISSUES]

- [PIMS-677: Create endpoint for LTSA API](https://apps.itsm.gov.bc.ca/jira/browse/PIMS-677)
- [PIMS-685: Create new unit tests for LTSA api](https://apps.itsm.gov.bc.ca/jira/browse/PIMS-685)

<br />

![CHECKLIST]

- [x] I have read and agree with the following checklist:

```
- I have performed a self-review of my own code.  
- I have commented my code, particularly in hard-to-understand areas.  
- I have made corresponding changes to the documentation where required.  
- I have tested my changes.  
- My changes generate no new warnings.  
```

[DESCRIPTION]: https://img.shields.io/badge/description-orange?style=for-the-badge
[TICKETS_AND_ISSUES]: https://img.shields.io/badge/tickets_&_issues-blue?style=for-the-badge
[CHECKLIST]: https://img.shields.io/badge/checklist-gold?style=for-the-badge

### CODE REVIEW

**Comment by BradyMitch** [comment]
@LawrenceLau2020 When I try to rebuild the backend I get

```
ERROR [backend build 16/20] RUN dotnet restore                                                                                  2.2s
------
 > [backend build 16/20] RUN dotnet restore:
#0 0.880 /usr/share/dotnet/sdk/7.0.305/NuGet.targets(398,5): error MSB3202: The project file "/src/ltsa/Pims.Ltsa.csproj" was not found. [/src/Pims.sln]
------
failed to solve: process "/bin/sh -c dotnet restore" did not complete successfully: exit code: 1
make[1]: *** [Makefile:65: build] Error 17
make[1]: Leaving directory 'C:/dev/BRAMITCH/PIMS-1'
make: *** [Makefile:68: rebuild] Error 2
```

**Comment by BradyMitch** [comment]
I tested in ThunderClient, well done!

File: backend/api/Areas/Tools/Controllers/LtsaController.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            _ltsaService = ltsaService;
+        }
+
+        [HttpGet("title")]
+        [HasPermission(Permissions.PropertyView)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(LtsaModel.LtsaOrderModel), 200)]
+        [SwaggerOperation(Tags = new[] { "ltsa" })]
+        public async Task<IActionResult> GetLandTitleInfo(string pid)
+        {
```
Comment (LawrenceLau2020) [inline backend/api/Areas/Tools/Controllers/LtsaController.cs:37]: The new endpoint is now: http://localhost:8081/api/ltsa/land/title?pid=xxxxxxxxx

**Comment by LawrenceLau2020** [comment]
> Tested on ThunderClient. Got 200 back as expected. All API unit test passing.
> 
> Good to merge, but I think we should think about returning our own error response in the future. Giving a bad pid just returns a 5xx, where you would expect maybe a 404 or 400.

Good catch! I've modified the controller to return the following JSON back if the call to the LTSA api doesn't work:
`{
  "message": "Unable to find title summary for parcel id: 028-770-90",
  "exception": "Failed to retrieve title summary for parcel id: 028-770-90. Status code: HTTP Request 'https://tduat-x42b.ltsa.ca/titledirect/search/api/titleSummaries?filter=parcelIdentifier:028-770-90' failed"
}`

**Comment by dbarkowsky** [comment]
> Good catch! I've modified the controller to return the following JSON back if the call to the LTSA api doesn't work

Very nice. Tested this out and it is working as expected. Feel free to merge.

**Pull request merged**